controllers/usersignupcleanup/usersignup_cleanup_controller.go

```diff
@@ -155,6 +184,52 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	return reconcile.Result{}, nil
 }
 
+func (r *Reconciler) isUserBanned(ctx context.Context, userSignup *toolchainv1alpha1.UserSignup) (bool, error) {
```

Comment:
I was wondering if checking the `Complete` condition would suffice, like we do in reg-service https://github.com/codeready-toolchain/registration-service/blob/1265c2318813f43cf0efa1c7bced960d1b0cffc1/pkg/signup/service/signup_service.go#L438-L445

or if there's anything different with host operator that we need to check also the banned resource and validate the email hash.

Reply:
Hey @mfrancisc , thanks for the review.  Unit tests have been added to the PR now.  We do a more thorough check to confirm that the user is banned in the host operator, as the BannedUser resource is the authorative source to determine whether a UserSignup is really banned.  Perhaps if the registration service is not checking this then it needs to be changed, it's been a long time since this code was written and I can't recall exactly why reg service was written this way.